pallets/domains/README.md

```diff
@@ -7,54 +7,59 @@ Domains pallet allows users to register and manage their domains. The main purpo
 
 The main storages of domains pallet are: 
 - `RegisteredDomains: Map<DomainName,DomainMeta>`\
-  Metadata associated with domain.
+  Metadata associated with a domain.
 - `DomainsByOwner: Map<AccountId, DomainName[]>`\
-  Domains owned by account.
+  Domains owned by an account.
+- `DomainRecords: Map<DomainName, Map<RecordKey, RecordValueWithDeposit>>`\
+  All records associated with a given domain name.
 
 
-## Domain metadata
+## Domain Records
 
 In addition to resolving a domain owner, you can also resolve the various records set up
-by the user. The following table shows a list of records that can be attached to the domain name.
-
-
-<table>
-<thead>
-  <tr>
-    <th>Record Name</th>
-    <th>Type</th>
-  </tr>
-</thead>
-<tbody>
-  <tr>
-    <td rowspan="3">Content</td>
-    <td>String</td>
-  </tr>
-  <tr>
-    <td>IPFS</td>
-  </tr>
-  <tr>
-    <td>Hypercore</td>
-  </tr>
-  <tr>
-    <td rowspan="3">InnerValue</td>
-    <td>Subsocial Account</td>
-  </tr>
-  <tr>
-    <td>Subsocial Space</td>
-  </tr>
-  <tr>
-    <td>Subsocial Post</td>
-  </tr>
-  <tr>
-    <td>OuterValue</td>
-    <td>String</td>
-  </tr>
-</tbody>
-</table>
+by the user.
+
+The domains record system allows any dotsama domain to have a key-value pairs of metadata attached to it.
+There is no restrictions on what value should be stored in what key, however we have a list of defined key schema to be 
+interpreted by clients.
+
+
+### Record Key standard
+
+A standard record key is split by namespaces by a . used as a separator.
+
+#### Crypto payment records
+
+A key of records regarding payment in crypto follows one of the following formats
+
+* `crypto.<TICKER>.address`
+  * `crypto.ETH.address` ⇒ `0xD1E5b0FF1287aA9f9A268759062E4Ab08b9Dacbe`
+  * `crypto.BTC.address` ⇒ `bc1qkd4um2nn2uyzmsch5y86wsa2pfh8xl445lg9nv`
+* `crypto.<TICKER>.version.<VERSION>.address`
+  * `crypto.USDT.version.ERC20.address` ⇒ `0x8aaD44321A86b170879d7A244c1e8d360c99DdA8`
+  * `crypto.USDT.version.TRON.address` ⇒ `THG9jVSMfKEbg4vYTYWjmLRyga3CKZdDsk`
+
+#### Browser resolution
+
+* `browser.redirect_url` ⇒ `http://example.com/home.html`
+
+#### Social records
+
+* `social.picture.value` ⇒ `ipfs://QmQqzMTavQgT4f4T5v6PWBp7XNKtoPmC9jvn12WPT3gkSE`
+* `whois.email.value` ⇒ `tarekkma@gmail.com`
+* `whois.for_sale.value` ⇒ `false`
+* `social.twitter.username` ⇒ `TarekkMA1`
+* `social.subsocial.account` ⇒ `3tiKakuy6RaHWThsHts23De8XwpaNdvw8PwwXcJqeVSN6w8w`
+* `social.subsocial.space` ⇒ `15`
+* `social.subsocial.post` ⇒ `1`
+
 
 ### Main types
 
+* `DomainName`: an array of characters representing a domain name (maximum length `Config::MaxDomainLength`)
+* `RecordKey`: an array of characters representing record key (maximum length `Config::MaxRecordKeySize`)
+* `RecordValue`: an array of characters representing record value (maximum length `Config::MaxRecordValueSize`)
```

Comment:
These descriptions sound too noisy. The names of the types are self-descriptive.
```suggestion
* `DomainName` – A byte array of maximum length of `Config::MaxDomainLength` bytes.
* `RecordKey` – A byte array of maximum length of `Config::MaxRecordKeySize` bytes.
* `RecordValue` – A byte array of maximum length of `Config::MaxRecordValueSize` bytes.
```